src/lib.rs

```diff
@@ -1484,6 +1484,17 @@ pub fn shallow_scan_for_device(
         .find(|&device| device.vendor_id == vendor_id && device.device_id == device_id)
 }
 
+pub fn get_pci_device_iter(
```

Comment:
vendor_id and device_id are unused here, what is the purpose? It seems like you're wanting maybe something like

```
all_local_devices(Bdf::from_bus(Bus(bus)), ops, Bus(bus + 1)).filter(|dev| dev.vendor_id == vendor_id && dev.device_id == device_id)
```
which is just adding a filter() call to all_local_devices(). does that need to be in this library? We have users of this library already that use it like:
```
        for dev in pci::all_local_devices(0, &ops)
            .filter(|dev| dev.vendor_id == RIVOS_VENDOR_ID && dev.device_id == IOMMU_DEVICE_ID)
```